apps/gitness/src/pages-v2/repo/repo-import-page.tsx

```diff
@@ -16,6 +16,20 @@ export const ImportRepo = () => {
   const { mutate: importRepoMutation, error, isLoading } = useImportRepositoryMutation({})
 
   const onSubmit = async (data: ImportRepoFormFields) => {
+    const providerRepo = data.provider.includes(ProviderOptionsEnum.GITHUB)
+      ? data.provider
+      : data.provider.includes(ProviderOptionsEnum.GITLAB)
+        ? data.group
+        : data.provider.includes(ProviderOptionsEnum.BITBUCKET)
+          ? data.workspace
+          : data.provider.includes(ProviderOptionsEnum.BITBUCKET_SERVER)
+            ? data.project
+            : data.provider.includes(ProviderOptionsEnum.GITEA) || data.provider.includes(ProviderOptionsEnum.GOGS)
+              ? data.organization
+              : data.provider.includes(ProviderOptionsEnum.AZURE_DEVOPS)
+                ? `${data.organization}/${data.project}`
+                : ''
```

Comment:
A bit hard to read and understand this maze like implementation, anything we can do better?

Reply:
make an map config of providers to supported fields, and use that to render conditionally

Reply:
done